src/Features/OnboardingTasks/TaskList.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Handles storage and retrieval of a task list
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\OnboardingTasks;
+
+/**
+ * Task List class.
+ */
+class TaskList {
+	/**
+	 * Option name hidden task lists.
+	 */
+	const HIDDEN_OPTION = 'woocommerce_task_list_hidden_lists';
+
+	/**
+	 * Option name completed task lists.
+	 */
+	const COMPLETED_OPTION = 'woocommerce_task_list_completed_lists';
+
+	/**
+	 * ID.
+	 *
+	 * @var string
+	 */
+	protected $id = '';
+
+	/**
+	 * Title.
+	 *
+	 * @var string
+	 */
+	protected $title = '';
+
+	/**
+	 * Title.
+	 *
+	 * @var array
+	 */
+	protected $tasks = array();
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $data Task list data.
+	 */
+	public function __construct( $data = array() ) {
+		$defaults = array(
+			'id'    => null,
+			'title' => '',
+			'tasks' => array(),
+		);
+
+		$data = wp_parse_args( $data, $defaults );
+
+		$this->id    = $data['id'];
+		$this->title = $data['title'];
+		$this->tasks = $data['tasks'];
+	}
+
+	/**
+	 * Check if the task list is hidden.
+	 *
+	 * @return bool
+	 */
+	public function is_hidden() {
+		$hidden = get_option( self::HIDDEN_OPTION, array() );
+		return in_array( $this->id, $hidden, true );
+	}
+
+	/**
+	 * Hide the task list.
+	 *
+	 * @return bool
+	 */
+	public function hide() {
+		$hidden   = get_option( self::HIDDEN_OPTION, array() );
+		$hidden[] = $this->id;
+		return update_option( self::HIDDEN_OPTION, array_unique( $hidden ) );
+	}
+
+	/**
+	 * Undo hiding of the task list.
+	 *
+	 * @return bool
+	 */
+	public function show() {
+		$hidden = get_option( self::HIDDEN_OPTION, array() );
+		$hidden = array_diff( $hidden, array( $this->id ) );
+		return update_option( self::HIDDEN_OPTION, $hidden );
+	}
+
+	/**
+	 * Check if the task list is complete.
+	 *
+	 * @return bool
+	 */
+	public function is_complete() {
+		// @todo This option is not yet being used to track completion.
```

Comment:
Is this todo necessary? The option may not be currently used, but once it is used, will we make sure we go back and remove this todo? It would be better to put todos where completion is currently tracked, so we know to update those.

Reply:
We can remove it. I'll create a follow-up issue to make sure we're tracking it.